src/anm/events.js

```diff
@@ -164,28 +164,31 @@ MouseEventsSupport.prototype.dispatch = function(event) {
         return true;
     }
 
-    var dispatchedChild;  // found the matching child inside
-    if (owner.inside(localEvent)) {
-        owner.reverseEach(function(child) {
-            if (child.isActive()) {
-                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
-                if (dispatchedChild) return false; // stop iteration of reverseEach
-            }
-        });
+    var dispatchedByOwner, // handled event myself
+        dispatchedByChild; // not handled myself, but found the matching child inside
 
-        if (!dispatchedChild) {
-            if (localEvent.type === 'mousemove') {
-                this.processMove(localEvent); // fire mouseenter/mouseexit if required
-            }
-            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
-                state.pressedNode = this.owner;
-            }
-            this.owner.fire(localEvent.type, localEvent);
+    // here and below localEvent has properties `.x` and `.y`, so duck typing works
+    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
```

Comment:
You asked me a question about checking bounds and dispatching to a child. I
answered. Yes, this was directed to you. We’ve discussed that so many times
and we have the spec that right about that.

On Tue, Jan 26, 2016 at 1:01 PM, Ulric Wilfred <notifications@github.com>
wrote:

> In src/anm/events.js
> <https://github.com/Animatron/player/pull/383#discussion_r50873484>:
>
> >
> > -        if (!dispatchedChild) {
> > -            if (localEvent.type === 'mousemove') {
> > -                this.processMove(localEvent); // fire mouseenter/mouseexit if required
> > -            }
> > -            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
> > -                state.pressedNode = this.owner;
> > -            }
> > -            this.owner.fire(localEvent.type, localEvent);
> > +    // here and below localEvent has properties `.x` and `.y`, so duck typing works
> > +    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
>
> @kirillk <https://github.com/kirillk> Also, in spec, there's no iteration
> through children in .contains method, same way it's done in PR, so seems
> your question "why on earth" was directed not to me?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Animatron/player/pull/383/files#r50873484>.
>